ui/selectors/transactions.js

```diff
@@ -103,8 +103,18 @@ export const smartTransactionsListSelector = (state) => {
     getCurrentChainId(state)
   ]
     ?.filter((stx) => {
-      const { txParams } = stx;
-      return txParams?.from === selectedAddress && !stx.confirmed;
+      const isCancelledSmartTransaction = stx.status?.startsWith('cancelled');
+      return (
+        stx.txParams?.from === selectedAddress &&
+        !stx.confirmed &&
+        (!isCancelledSmartTransaction ||
+          // We only want to show cancelled Smart Transactions for Swaps in Activity,
```

Comment:
In the future, would it be cleaner to just load all transactions from the `TransactionController` since it's the source of truth for transaction state and all new smart transactions (excluding swaps) are created from it?

We could still use the smart transaction state to augment the displayed transaction data, but we wouldn't need to load in additional transactions entirely.

Reply:
Agreed. It's one of the post-MVP user stories to unify behaviour between non-swaps and swaps smart transactions.